src/mon/PGMonitor.cc

```diff
@@ -535,6 +578,10 @@ bool PGMonitor::preprocess_query(MonOpRequestRef op)
 
 bool PGMonitor::prepare_update(MonOpRequestRef op)
 {
+  if (mon->osdmon()->osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
+    return false;
```

Comment:
@liewegas, there's some haphazard quick returns on invoking pgmon post-luminous, but shouldn't we be doing something like removing it from the dispatch loops entirely?
I'd actually think we want to just not load it into memory at all once we're past the upgrade point, and maybe even unload it, but I haven't looked into how tricky that is yet....